### PR TITLE
Rafael/dataperf changes

### PR DESCRIPTION
**Pull request by remg1997**

### CODE REVIEW

File: api/controllers/models.py [hunk 1]
```diff
@@ -6,13 +6,20 @@
...
 import tempfile
 import time
+from typing import List
 
 import boto3
 import bottle
+import numpy as np
+import pandas as pd
+import requests
+import sklearn
```
[the remark this answers is not in the document]
Comment (remg1997) [reply api/controllers/models.py:22]: Yes! I didn't include the built-in ones.

File: api/controllers/models.py [hunk 2]
```diff
@@ -244,28 +223,153 @@ def do_upload_via_train_files(credentials, tid, model_name):
...
-                    # this consistent?
-                    del datum["uid"]
-                    tmp.write(util.json_encode(datum) + "\n")
-                tmp.close()
-                ret = _eval_dataset(dataset_name, endpoint_name, model, task, tmp.name)
-                status_dict.update(ret)
+
+    for name, upload in train_files.items():
+
+        old_metadata = sm.getByTid(27)[0].metadata_json
```
[the remark this answers is not in the document]
Comment (remg1997) [reply api/controllers/models.py:229]: Oh this isn't required anymore. Thanks.

File: api/controllers/models.py [hunk 3]
```diff
@@ -244,28 +223,153 @@ def do_upload_via_train_files(credentials, tid, model_name):
...
+
+        for id, label in current_upload.items():
+            id_count += 1
+            try:
+                train_X.append(list(get_train_dataset_embedding(id)))
+                train_y.append(label)
+            except:
+                continue
+
+        if id_count > 500:
```
[the remark this answers is not in the document]
Comment (remg1997) [reply api/controllers/models.py:247]: Right. Setting it up to 1000, as discussed with the vision team.

File: api/controllers/models.py [hunk 4]
```diff
@@ -244,28 +223,153 @@ def do_upload_via_train_files(credentials, tid, model_name):
...
+            did=did,
+            pretty_perf=f"{f1_score} %",
+            perf=f1_score,
+            metadata_json=new_score_string,
+        )
+
+        # accumulated_predictions += predictions['predictions'] Implementation for accumulated labels instead of mean
+        # accumulated_labels += test_y Implementation for accumulated labels instead of mean
+
+    # f1_score = sklearn.metrics.f1_score(accumulated_labels, accumulated_predictions) Implementation for accumulated labels instead of mean
```
[the remark this answers is not in the document]
Comment (remg1997) [reply api/controllers/models.py:318]: Dynabench currently calculates the average for each of the testing datasets to determine the pooled f1-score. Nevertheless, I wanted to discuss the possibility of having two arrays containing all predictions and ground truths (for all datasets) and calculating a single f1-score as a pooling method. I'm still not sure if the result is equivalent to the previous implementation, therefore I implemented it but left it as a possibility.

**Review by Ciroye — APPROVED** [review]

**Pull request merged**